modules/cloudngfw/main.tf

```diff
@@ -0,0 +1,105 @@
+data "aws_caller_identity" "current" {}
+
+resource "cloudngfwaws_ngfw" "this" {
+  name        = var.name
+  vpc_id      = var.vpc_id
+  account_id  = data.aws_caller_identity.current.id
+  description = var.description
+
+  endpoint_mode = var.endpoint_mode
+
+  dynamic "subnet_mapping" {
+    for_each = var.subnets
+
+    content {
+      #subnet_id = try(subnet_mapping.value.id, null)
+      availability_zone = subnet_mapping.key
+    }
+  }
+
+  rulestack = cloudngfwaws_commit_rulestack.this.rulestack
+
+  tags = var.tags
+}
+
+resource "cloudngfwaws_commit_rulestack" "this" {
+  rulestack = cloudngfwaws_rulestack.this.name
+}
+
+resource "cloudngfwaws_rulestack" "this" {
+  name  = var.rulestack_name
+  scope = var.rulestack_scope
+
+  account_id  = data.aws_caller_identity.current.id
+  description = var.description_rule
+
+  profile_config {
+    anti_spyware  = var.profile_config["anti_spyware"]

```

Comment:
Variable `profile_confi` is defined as `map(any)` and it doesn't contain any validations, so if e.g. `anti_spyware` doesn't exists in the map, we have an error . Maybe we could use it here `try()` for handling errors ?